app/controlplane/internal/service/workflow.go

```diff
@@ -103,13 +105,33 @@ func (s *WorkflowService) Update(ctx context.Context, req *pb.WorkflowServiceUpd
 	return &pb.WorkflowServiceUpdateResponse{Result: bizWorkflowToPb(p)}, nil
 }
 
-func (s *WorkflowService) List(ctx context.Context, _ *pb.WorkflowServiceListRequest) (*pb.WorkflowServiceListResponse, error) {
+func (s *WorkflowService) List(ctx context.Context, req *pb.WorkflowServiceListRequest) (*pb.WorkflowServiceListResponse, error) {
 	currentOrg, err := requireCurrentOrg(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	workflows, err := s.useCase.List(ctx, currentOrg.ID)
+	// If project reference is provided, find the project and scope the workflow List request
+	// to only projects within the project
+	var projectID string
+	if req.GetProjectReference().GetRef() != nil {
+		pro, err := s.projectsUseCase.FindProjectByReference(ctx, currentOrg.ID, &biz.EntityRef{
+			ID:   req.GetProjectReference().GetEntityId(),
+			Name: req.GetProjectReference().GetEntityName(),
+		})
+		if err != nil {
+			return nil, handleUseCaseErr(err, s.log)
+		}
+
+		// If the project is not found, return an empty response
+		if pro == nil {
+			return &pb.WorkflowServiceListResponse{}, nil
```

Comment:
return 404 instead? Why return an empty list? I am not sure what's the right way of doing this, just curious

Reply:
I can see 404 if the workflow is not found when filtering runs?

```
chainloop wf run ls --workflow foo --project bar
ERR failed to get workflow: workflow not found
```